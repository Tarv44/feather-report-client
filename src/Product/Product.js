import React, { Component } from 'react';
import styles from './Product.module.css';
import ProductAction from '../ProductAction/ProductAction';
import EditProduct from '../EditProduct/EditProduct';

export default class Product extends Component {
    state = {
        type: this.props.type,
        editing: false,
        selected: false
    }

    displayEdit = (e) => {
        e.preventDefault()
        this.setState({ editing: true })
    }

    toggleSelected = () => {
        this.setState({ selected: !this.state.selected })
    }

    render() {
        const product = this.props.productData

        const features = this.state.type === 'compare'
            ? this.props.allFeatures.map((f, i) => {
                for (let l = 0; l < product.features.length; l++) {
                    if (product.features[l].id === f.id) {
                        return <li key={i} id={f.id}>{f.message}</li>
                    }
                }
                return <li key={i} id={f.id} className={styles.notFeatured}>{f.message}</li>
            })
            : product.features.map((f, i) => {
                if (f === undefined) {
                    return <li key={i}>Loading...</li>
                }
                return <li key={i}>{f.message}</li>
            })

        
        const action = this.state.type === 'edit'
            ? <ProductAction 
                handleForm={this.displayEdit} 
                type={this.state.type} 
            />
            : this.state.type === 'select'
            ? <ProductAction 
                handleSelected={this.props.handleSelected}
                product={product}
                selected={this.state.selected} 
                type={this.state.type} 
            />
            : <ProductAction 
                type={this.state.type} 
            />

        return this.state.editing 
            ? <EditProduct product={product}/>
            : (
                <div className={styles.product}>
                    <div className={styles.content}>
                        <h2>{product.title}</h2>
                        <p>${product.price}</p>
                    </div>
                    
                    <p>{product.description}</p>
                    <div className={styles.content}>
                        <ul>
                            {features}
                        </ul>
                        {action}
                    </div>
                </div>
            )
    }
}