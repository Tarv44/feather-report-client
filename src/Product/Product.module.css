.product {
    border: solid black 3px;
    padding: 5px 10px;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}

.product_content {
    max-width: 100% !important;
}

.notFeatured {
    text-decoration: line-through;
    opacity: .3;
}