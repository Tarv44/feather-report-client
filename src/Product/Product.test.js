import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import Product from './Product'
import { product, categories } from '../component.fixtures'

describe('----- Product -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
            <Product 
                productData={product}
                type={'edit'}
                allCats={categories}
            />
        , div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <Product 
                    productData={product}
                    type={'edit'}
                    allCats={categories}
                />
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})