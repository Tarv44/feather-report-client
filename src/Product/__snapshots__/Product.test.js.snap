// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- Product ----- renders the UI as expected 1`] = `
<div
  className="product"
>
  <div
    className="content"
  >
    <h2>
      Sunbeam Wide Slot 4-Slice Toaster
    </h2>
    <p>
      $
      24.99
    </p>
  </div>
  <p>
    Get perfectly toasted food every time! The Sunbeam 4-Slice Toaster uses electronic toasting-technology for uniformly toasted bread, bagels, English muffins and more, just the way you like.
  </p>
  <div
    className="content"
  >
    <ul>
      <li>
        4-Slice
      </li>
      <li>
        Extra-Wide Slot
      </li>
      <li>
        Black
      </li>
      <li>
        Removable Crumb Tray
      </li>
      <li>
        Anti-Jam Shut-Off
      </li>
    </ul>
    <button
      className="button"
      onClick={[Function]}
    >
      Edit Product
    </button>
  </div>
</div>
`;
